lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1909,7 +1935,7 @@ sub blocked_by_parent_job {
 
     my $chained_parents = $self->result_source->schema->resultset('JobDependencies')->search(
         {
-            dependency   => OpenQA::JobDependencies::Constants::CHAINED,
+            dependency   => {-in => [OpenQA::JobDependencies::Constants::CHAINED_DEPENDENCIES]},
```

Comment:
a tricky one. Make sure we cover the case in test: If A is non-blocked, it's direct siblings need to be unblocked as well. The scheduler will not even look at blocked jobs.

Reply:
My idea was to pull directly chained dependencies at the point the scheduler so far only assigns one job to a worker. It would then check whether this job has "fanboys" (as you called them once) and those fanboys would be added to the assignment regardless of their blocked-by state (of course checking for misconfigurations like cycles and compatible worker class). So they would be even blocked while in the running state. I guess as soon as the worker actually works on them (sends status updates) the web UI would remove the blocked by.

Reply:
As long as the fanboys are in the cluster information, we should be fine. 

But thinking: having 1000 jobs in 'running tests' just because the first started sounds far from optimal. So I would rather like to see them in scheduled/blocked - even if the worker already got the information what to work on.

Reply:
We can also keep the jobs which are already accepted by a worker but still not actually running in scheduled/blocked. It shouldn't be hard to prevent the scheduler from ignoring them.

Reply:
as I said: if they are blocked, scheduler ignores them. He would also need to ignore basically all fanboys

Reply:
I'm resolving the conversation without modifying the code because the following mentioned points should be true/implementable with this behavior:

* As long as the fanboys are in the cluster information, we should be fine.
* So I would rather like to see them in scheduled/blocked - even if the worker already got the information what to work on.

And I keep "He would also need to ignore basically all fanboys" in mind.